### DISCUSSION TITLE
Webterm future

### DISCUSSION THREAD

**Discussion by matthewauld** [beakerbrowser/beaker#1750] [General] [answered]
I know that (for entirely good reasons) webterm isn't at the top of the priority lists development-wise, but I was wondering what/if the future plans are for it? It seems like it could become a powerful tool that could real advantages over a traditional command line like the ability to return javascript objects, numbers and strings from commands, and can provide rich output. It kind of grabbed my imagination.

I was wondering what people thought about some of the following ideas, or there were better ways to do these things?

- composing commands with `then` and `catch` (e.g. readFile xxx json `then` filter x.id==1 `then` echo `catch` echo "not a file"). commands that wanted to allow composing could take a `input=null` argument. 
- exposing webterm command back-ends to one another through the `this` object- so when programming a custom command, being able to access the functions of other commands to compose them together. It would create some dependency issues, but nothing that could be worked out.
- scripts: one of the huge advantages of a command line  is the ability to automate a process. I don't see an advantage to building a new scripting language (since javascript is already a good one that people know), but being able to run "javascript" that could run and have access to the `this` object of webterm might be a compromise.

**Comment by pfrazee** [comment, marked as answer]
Hey @matthewauld yeah it's awesome to hear you're digging Webterm.

I'm working right now (half a commit done) to sandbox webterm commands properly so we can feel less anxiety about running them. That seems like a pretty key first step, so it wont be in this week's release but probably the next one.

There's a semi-outdated [webterm specs repo](https://github.com/beakerbrowser/webterm). Why don't we start doing proposals there? I think command composition would be a great thing to start discussing.

**Reply by matthewauld** [reply]
Oh, neat, I didn't realize it had all be laid out.

**Reply by pfrazee** [reply]
Yeah we're ready to go!

**Reply by matthewauld** [reply]
For anyone reading this - [this](https://github.com/beakerbrowser/webterm/issues/4) is a way better way to do composition, as it can be non-linear.

**Comment by johanbove** [comment]
I love webterm. I use my "blogpost" command to write my microblog posts and it's to me at least, the fastest way to browse through drives.

**Reply by matthewauld** [reply]
That is a great idea. Honestly never thought of doing that!
